hubspot/tasks.py

```diff
@@ -68,16 +74,67 @@ def check_hubspot_api_errors():
         defaults={"checked_on": now_in_utc()}
     )
     last_timestamp = hubspot_timestamp(last_check.checked_on)
+
     for error in get_sync_errors():
-        if error.get("errorTimestamp") > last_timestamp:
-            msg = "Hubspot error for {obj_type} id {obj_id}: {details}".format(
-                obj_type=error.get("objectType", "N/A"),
-                obj_id=error.get("integratorObjectId", "N/A"),
-                details=error.get("details", ""),
+        error_timestamp = error.get("errorTimestamp")
+        if error_timestamp > last_timestamp:
+            obj_type = (error.get("objectType", "N/A"),)
+            obj_id = parse_hubspot_id(error.get("integratorObjectId", ""))
+            error_type = error.get("type", "N/A")
+            details = error.get("details", "")
+
+            if (
+                obj_id is not None
+                and "LINE_ITEM" in obj_type
+                and error_type == "INVALID_ASSOCIATION_PROPERTY"
+                and retry_invalid_line_association(obj_id, error_timestamp, details)
+            ):
+                continue
+
+            log.error(
+                "Hubspot error %s for %s id %s: %s",
+                error_type,
+                obj_type,
+                str(obj_id),
+                details,
             )
-            log.error(msg)
         else:
             break
 
     last_check.checked_on = now_in_utc()
     last_check.save()
+
+
+def retry_invalid_line_association(line_id, error_timestamp, details):
+    """
+    Handle an INVALID ASSOCIATION PROPERTY on a LINE_ITEM object.
+
+    Args:
+        line_id (int): ID of the line object parsed from the error json
+        error_timestamp (int): timestamp when error occurred
+        details (str): details string containing order_id
+    Returns:
+        (bool): True if the error was handled in some way, False if not
+    """
+    order_id = parse_hubspot_id(ASSOCIATED_DEAL_RE.search(details).group(1))
+    try:
+        line = Line.objects.get(id=line_id)
+    except ObjectDoesNotExist:
+        return False
+
+    try:
+        # Get sync data and cache it to check against potential other line errors
+        sync_data = get_sync_status("DEAL", line.order.id)
+    except HTTPError as sync_status_error:
+        if sync_status_error.response.status_code == 400:
+            # 400 status indicates order is pending, ignore the error for now
+            return True
+        else:
+            log.error(sync_status_error)
+            return True
+
+    if order_id != line.order.id or sync_data["lastProcessedAt"] > error_timestamp:
+        # If the line's order has changed or the order has synced since the error, resync the line
+        sync_line_item_with_hubspot.si(line.id)
```

Comment:
Seems like celery never actually gets around to running this, I think it may be because it's getting called within another task. Getting rid of the `.si` seems to work locally:
```python
sync_line_item_with_hubspot(line_id)
```